src/main/java/org/wildfly/security/http/impl/DigestAuthenticationMechanism.java

```diff
@@ -327,7 +328,15 @@ private void prepareResponse(String realmName, HttpServerResponse response, bool
     }
 
     public byte[] getPredigestedSaltedPassword(RealmCallback realmCallback, NameCallback nameCallback, String passwordAlgorithm, String mechanismName) throws AuthenticationMechanismException {
-        CredentialCallback credentialCallback = new CredentialCallback(PasswordCredential.class, passwordAlgorithm);
+        final String realmName = realmCallback.getDefaultText();
+        final String userName = nameCallback.getDefaultName();
+        final DigestPasswordAlgorithmSpec parameterSpec;
+        if (realmName != null && userName != null) {
+            parameterSpec = new DigestPasswordAlgorithmSpec(userName, realmName);
```

Comment:
This is incorrect - it requests URP digest for default userName (authzid), not for authentication name (from NameCallback) - will fix as part of https://issues.jboss.org/browse/ELY-991